openapi/openapi/src/main/scala/endpoints4s/openapi/Urls.scala

```diff
@@ -48,13 +48,39 @@ trait Urls extends algebra.Urls {
       )
     )
 
-  case class DocumentedQueryStringParam(schema: Schema, isRequired: Boolean)
-  type QueryStringParam[A] = DocumentedQueryStringParam
+  type WithDefault[A] = A
+
+  def optQsWithDefault[A](name: String, default: A, docs: Documentation = None)(implicit
+      value: QueryStringParam[A]
+  ): QueryString[WithDefault[A]] =
+    DocumentedQueryString(
+      List(
+        DocumentedParameter(
+          name,
+          required = false,
+          docs,
+          schema = {
+            val defaultJson = value.encoder.encode(default)
+            value.schema.withDefinedDefault(defaultJson)
+          }
+        )
+      )
+    )
+
+  case class DocumentedQueryStringParam[A](
+      schema: Schema,
+      isRequired: Boolean,
+      encoder: Encoder[A, Option[ujson.Value]]
```

Comment:
I think it should be
```suggestion
      encoder: Encoder[A, ujson.Value]
```

And then in `optQsWithDefault` you would write `value.schema.withDefinedDefault(Some(defaultJson))`

Reply:
Optional json value here is not defined because of `optQsWithDefault`, but to avoid [this issue](https://github.com/endpoints4s/endpoints4s/pull/879#discussion_r729649560) by basically not encoding any value in case of `None` for `optionalQueryStringParam`.

Reply:
Right, thanks for explaining it again :) Do you mind adding a scaladoc comment explaining that?

Reply:
I hope I made it somewhat clearer 🙂